nautobot/extras/models/jobs.py

```diff
@@ -278,13 +280,13 @@ def git_repository(self):
 
     @property
     def job_task(self):
-        """Get the registered Celery task, refreshing it if necessary."""
-        if self.git_repository is not None:
-            # If this Job comes from a Git repository, make sure we have the correct version of said code.
-            refresh_git_repository(
-                state=None, repository_pk=self.git_repository.pk, head=self.git_repository.current_head
-            )
-        return app.tasks[f"{self.module_name}.{self.job_class_name}"]
+        """Get an instance of the associated Job class, refreshing it if necessary."""
+        from nautobot.extras.jobs import get_job
+
+        try:
+            return get_job(self.class_path, reload=True)()
+        except TypeError as err:  # keep 2.0-2.2.1 exception behavior
```

Comment:
Should this be `2.0-2.2.2`?
```suggestion
        except TypeError as err:  # keep 2.0-2.2.2 exception behavior
```